src/service/icon_generator/icon_generator.service.ts

```diff
@@ -0,0 +1,116 @@
+'use strict';
+import * as Jimp from 'jimp';
+import * as crypto from 'crypto';
+import { Complex } from 'src/shared/util/mathUtils';
+
+const imageSize = 1028;
+const halfSize = imageSize / 2;
+
+const colorElementNumber = 4;
+const emptyColor = [240, 240, 240, 255];
+
+const colorRange = 256;
+const minColor = 0;
+const maxColor = colorRange - 1;
+
+const minColorIntensity = 70;
+const maxColorIntensity = 230;
+const magicRandomNumber = 10005;
+
+const juliaSetThreshold = 10;
+const juliaSetMaxIter = 20;
+
+const hashDigitsToAccount = 10;
+const hex = 16;
+
+const scaleMultiplier = 2 / imageSize;
+
+export class ImageGen {
+  public static generateImage(userEmail: string): Promise<Buffer> {
+    const hash = ImageGen.generateHash(userEmail);
+
+    const img = new Jimp(imageSize, imageSize);
+
+    const colorSeed = parseInt(hash.substring(0, hashDigitsToAccount));
+    const randGen = ImageGen.random(colorSeed);
+
+    const filledColor = ImageGen.getRandomColor(randGen);
+
+    const JuliaSetPointOffset = new Complex(ImageGen.toParamRange(randGen()), ImageGen.toParamRange(randGen()));
```

Comment:
Create an intermediate variable for `ImageGen.toParamRange(randGen())`

Reply:
It is impossible. RandGen should retrieve different results when you call it. I can't create one variable for 2 different values